stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -586,6 +586,22 @@ extension ${Self} {
     fatalError("unavailable function can't be called")
   }
 }
+
+extension Int {
+  @_transparent
+  @warn_unused_result
+  public init<U>(bitPattern: ${Self}<U>) {
```

Comment:
What would be the correct behavior of a null pointer? Setting the Int/UInt to 0, or making this a failable initializer and returning nil?